rs/tests/boundary_nodes/rate_limit_canister_test.rs

```diff
@@ -0,0 +1,390 @@
+            },
+            // Only this rule is different from config 1.
+            // It means that the old rule is removed (not mutated) and this new rule is applied instead.
+            InputRule {
+                incident_id: "incident_id_4".to_string(),
+                rule_raw: rule_3.to_bytes_json().unwrap(),
+                description: "Some vulnerability #4 discovered".to_string(),
+            },
+            InputRule {
+                incident_id: "incident_id_3".to_string(),
+                rule_raw: rule_4.to_bytes_json().unwrap(),
+                description: "Some vulnerability #3 discovered".to_string(),
+            },
+        ],
+    })
+    .unwrap();
+
+    let result = agent
+        .update(&canister_id, "add_config")
+        .with_arg(args)
+        .call_and_wait()
+        .await
+        .unwrap();
+
+    let decoded = Decode!(&result, AddConfigResponse).unwrap();
+
+    info!(&logger, "Response to add_config() call: {decoded:#?}");
+}
+
+async fn read_config(
+    logger: Logger,
+    agent: &Agent,
+    version: Version,
+    canister_id: Principal,
+) -> Vec<RuleId> {
+    let args = Encode!(&Some(version)).unwrap();
+
+    let response = agent
+        .update(&canister_id, "get_config")
+        .with_arg(args)
+        .call_and_wait()
+        .await
+        .expect("update call failed");
+
+    let decoded = Decode!(&response, GetConfigResponse)
+        .expect("failed to decode candid response")
+        .unwrap();
+
+    info!(&logger, "Response to get_config() call: {}", decoded);
+
+    decoded
+        .config
+        .rules
+        .into_iter()
+        .map(|rule| rule.id)
+        .collect()
+}
+
+async fn disclose_incident(
+    logger: Logger,
+    agent: &Agent,
+    incident_id: IncidentId,
+    canister_id: Principal,
+) {
+    let disclose_arg = DiscloseRulesArg::IncidentIds(vec![incident_id]);
+    let args = Encode!(&disclose_arg).unwrap();
+
+    let response = agent
+        .update(&canister_id, "disclose_rules")
+        .with_arg(args)
+        .call_and_wait()
+        .await
+        .expect("update call failed");
+
+    let decoded = Decode!(&response, DiscloseRulesResponse).unwrap();
+
+    info!(&logger, "Response to disclose_rules() call: {decoded:#?}");
+}
+
+async fn read_rule(
+    logger: Logger,
+    agent: &Agent,
+    rule_id: &RuleId,
+    canister_id: Principal,
+) -> OutputRuleMetadata {
+    let args = Encode!(rule_id).unwrap();
+
+    let response = agent
+        .update(&canister_id, "get_rule_by_id")
+        .with_arg(args)
+        .call_and_wait()
+        .await
+        .expect("update call failed");
+
+    let rule_metadata = Decode!(&response, GetRuleByIdResponse).unwrap().unwrap();
+
+    info!(
+        &logger,
+        "Response to get_rule_by_id() call: {rule_metadata}"
+    );
+
+    rule_metadata
+}
+
+fn main() -> Result<()> {
+    SystemTestGroup::new()
+        .with_setup(config)
```

Comment:
Could you rename this to the more modern:
```suggestion
        .with_setup(setup)
```